redact_sample.py

```diff
@@ -0,0 +1,117 @@
+
+from generate_sample_file import SampleGenerator
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.NullHandler())
+SampleGen = SampleGenerator()
+SampleGen.ESTAB_TYPES = ['HALL OF RESIDENCE', 'CARE HOME', 'HOSPITAL', 'HOSPICE', 'MENTAL HEALTH HOSPITAL',
+                         'MEDICAL CARE OTHER', 'BOARDING SCHOOL', 'HOTEL',
+                         'YOUTH HOSTEL', 'HOSTEL', 'EDUCATION OTHER', 'PRISON', 'STAFF ACCOMMODATION', 'CAMPHILL',
+                         'HOLIDAY PARK', 'HOUSEHOLD', 'RESIDENTIAL CARAVAN', 'RESIDENTIAL BOAT', 'GATED APARTMENTS',
+                         'FOREIGN OFFICES', 'CASTLES', 'EMBASSY', 'CARAVAN', 'MARINA']
+SampleGen.read_words()
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(description='Load a sample file into response management.')
+    parser.add_argument('sample_file_path', help='path to the sample file', type=str)
+    return parser.parse_args()
+
+
+def redact_sample_file(sample_file_path, output_file_path):
+    with open(sample_file_path) as sample_file:
+        redact_sample(sample_file, output_file_path)
+    sample_file.close()
+
+
+def redact_sample(sample_file: Iterable[str], output_file_path):
+    sample_file_reader = csv.DictReader(sample_file, delimiter=',')
+    _redact_sample_units(sample_file_reader, output_file_path)
+
+
+def _redact_sample_units(sample_file_reader: Iterable[str], output_file_path,
+                         sample_unit_log_frequency=50000):
+    logger.info('Redacting sample...')
+
+    with open(output_file_path, 'w', newline='') as output_file:
+        writer = csv.DictWriter(output_file, fieldnames=SampleGenerator.FIELDNAMES)
+        writer.writeheader()
+
+        for count, sample_row in enumerate(sample_file_reader, 1):
+            redacted_sample_row = _redact_sample_row(sample_row)
+            _write_row(writer, redacted_sample_row)
+
+            if count % sample_unit_log_frequency == 0:
+                logger.info(f'{count} sample units redacted')
+
+    logger.info('All sample units have been redacted')
+
+
+def _redact_sample_row(sample_row):
+    sample_row['HTC_WILLINGNESS'] = SampleGen.get_random_htc()
+    sample_row['HTC_DIGITAL'] = SampleGen.get_random_htc()
+    if sample_row['ESTAB_TYPE'] in ['MILITARY US SFA', 'MILITARY SFA', 'MILITARY US SLA', 'ROYAL HOUSEHOLD',
+                                    'MOD HOUSEHOLDS', 'HIGH SECURE MENTAL HEALTH', 'ROUGH SLEEPER', 'TRANSIENT PERSONS',
+                                    'TRAVELLING PERSONS', 'GRT SITE', 'MIGRANT WORKERS', 'IMMIGRATION REMOVAL CENTRE',
+                                    'SHELTERED ACCOMMODATION', 'APPROVED PREMISES', 'RESIDENTIAL CHILDRENS HOME',
+                                    'RELIGIOUS COMMUNITY', 'LOW/MEDIUM SECURE MENTAL HEALTH']:
+        sample_row['ESTAB_TYPE'] = SampleGen.get_random_estab_type()
+        sample_row['ADDRESS_LINE1'] = SampleGen.get_random_address_line()
+        sample_row['ADDRESS_LINE2'] = ''
```

Comment:
This is a hangover from more than a year ago, when I hacked together the original sample generator. We really would like to have random address line 2 & 3 too, plus maybe some random mix between addresses which have 1, 2 and 3 lines, respectively

Reply:
Happy to implement, are address lines 2 and 3 okay to be random words as if they were village names etc?

Reply:
yes sure, as discussed